compiler/compiler_test.go

```diff
@@ -7,49 +7,35 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/testing/protocmp"
 	pb "google.golang.org/protobuf/types/descriptorpb"
 )
 
-func requireNoError(t *testing.T, err error) {
+func requireProtoEqual(t *testing.T, want, got proto.Message) {
```

Comment:
Get rid of this and just use `require.Equal()`

Reply:
can't sorry. `require.Equal()` doesn't return true when to protomessages are equal, and JSONEq is too sketchy imo.
also, the output is more parseable with what I have here imo:

It is:
```
            	            	expected: &descriptorpb.FileDescriptorSet{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(0xc0001ca000)}, sizeCache:0, unknownFields:[]uint8(nil), File:[]*descriptorpb.FileDescriptorProto{(*descriptorpb.FileDescriptorProto)(0xc000352e00)}}
            	            	actual  : &descriptorpb.FileDescriptorSet{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(0xc0001ca000)}, sizeCache:0, unknownFields:[]uint8(nil), File:[]*descriptorpb.FileDescriptorProto{(*descriptorpb.FileDescriptorProto)(0xc000352b00)}}
  ```
  versus
  ```              (*descriptorpb.FileDescriptorSet)(Inverse(protocmp.Transform, protocmp.Message{
              	"@type": s"google.protobuf.FileDescriptorSet",
              	"file": []protocmp.Message{
              		{"@type": s"google.protobuf.FileDescriptorProto", "message_type": []protocmp.Message{{"@type": s"google.protobuf.DescriptorProto", "field": []protocmp.Message{{"@type": s"google.protobuf.FieldDescriptorProto", "json_name": string("s"), "label": s"LABEL_OPTIONAL", "name": string("s"), ...}}, "name": string("M3")}}, "name": string("01_proto2_pkg.proto"), "package": string("pkg")},
              		{
              			"@type": s"google.protobuf.FileDescriptorProto",
              			"message_type": []protocmp.Message{
              				{
              					"@type": s"google.protobuf.DescriptorProto",
              					"field": []protocmp.Message{
              						{
              							"@type":     s"google.protobuf.FieldDescriptorProto",
            - 							"json_name": string("aDouble"),
            + 							"json_name": string("aDoubles"),
              							"label":     s"LABEL_OPTIONAL",
            - 							"name":      string("a_double"),
            + 							"name":      string("a_doubles"),
  ```

Reply:
Still an open issue it seems:
https://github.com/stretchr/testify/issues/758